tools/depscan/main.go

```diff
@@ -0,0 +1,302 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"debug/buildinfo"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+)
+
+func main() {
+	rootCmd := buildRootCommand()
+
+	err := rootCmd.ExecuteContext(context.Background())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func buildRootCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "depscan",
+	}
+
+	cmd.AddCommand(buildLicenseCommand())
+
+	klog.InitFlags(nil)
+
+	return cmd
+}
+
+type Module struct {
+	Name string `json:"name"`
+	Sum  string `json:"sum"`
+	//Path    string
+	Version string `json:"version"`
+
+	Info         *moduleInfo   `json:"licenseInfo,omitempty"`
+	LicenseFiles []LicenseFile `json:"licenseFiles,omitempty"`
+}
+
+func buildLicenseCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "license",
+	}
+
+	cmd.AddCommand(buildLicenseScanCommand())
+
+	return cmd
+}
+
+func buildLicenseScanCommand() *cobra.Command {
+	var opts RunLicenseScanOptions
+
+	cmd := &cobra.Command{
+		Use: "scan",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return RunLicenseScan(cmd.Context(), opts)
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.Binary, "binary", opts.Binary, "binary to analyze")
+
+	cmd.Flags().BoolVar(&opts.IncludeLicenses, "print", opts.IncludeLicenses, "include license text")
+
+	cmd.Flags().StringArrayVar(&opts.IgnorePackage, "ignore", opts.IgnorePackage, "packages to ignore")
+
+	return cmd
+}
+
+type RunLicenseScanOptions struct {
+	Binary string
+
+	IgnorePackage []string
+
+	IncludeLicenses bool
+}
+
+func RunLicenseScan(ctx context.Context, opts RunLicenseScanOptions) error {
+	buildInfo, err := buildinfo.ReadFile(opts.Binary)
+	if err != nil {
+		return fmt.Errorf("error reading binary info from %q: %w", opts.Binary, err)
+	}
+
+	var errors []error
+
+	var modules []*Module
+	for _, dep := range buildInfo.Deps {
+		for _, ignorePackage := range opts.IgnorePackage {
```

Comment:
curious about ignoring package. What led you to add support for this ?

Reply:
Added a note - it's to avoid internal libraries when they are detected as modules.